docs/userguide.rst

```diff
@@ -0,0 +1,204 @@
+.. _userguide:
+
+User guide
+==========
+
+Prerequisites
+-------------
+
+Currently reana-cluster command line tool supports deploying
+REANA components only to `Kubernetes <https://kubernetes.io/docs/concepts/overview/what-is-kubernetes/>`_
+infrastructure, so in order to deploy REANA components one needs to have a
+working Kubernetes deployment.
+
+reana-cluster utilizes configuration files of `kubectl -tool <https://kubernetes.io/docs/user-guide/kubectl-overview/>`_
+to set up a connection to Kubernetes infrastructure, so in order
+to use reana-cluster one needs to have kubectl installed
+and configured properly.
+
+
+Test reana-cluster using Minikube
+---------------------------------
+
+Easiest way to test reana-cluster without dedicating time to
+setting up a real Kubernetes infrastructure is to use
+`Minikube <https://kubernetes.io/docs/getting-started-guides/minikube/>`_,
+which is a spins up a working Kubernetes deployment in a virtual machine.
+
+Once you have installed kubectl and Minikube, start Minikube by running:
+
+.. code-block:: console
+
+   $ minikube start --kubernetes-version="v1.6.4"
+   ...
+   minikube: Running
+   cluster: Running
+   kubectl: Correctly Configured: pointing to minikube-vm at 192.168.99.100
+
+.. note::
+   Currently REANA supports only Kubernetes v1.6.4
+
+As seen from the output, Minikube startup routines should
+configure kubectl to interact with the newly created
+virtual machine, but it best to test that kubectl is indeed
+configured properly:
+
+.. code-block:: console
+
+   $ kubectl get all
+   NAME             CLUSTER-IP   EXTERNAL-IP   PORT(S)   AGE
+   svc/kubernetes   10.0.0.1     <none>        443/TCP   3d
+
+
+Install reana-cluster cli tool
+------------------------------
+
+reana-cluster command line interface tool is not yet released
+in PyPI, so install it from our GitHub repotory:
+
+.. code-block:: console
+
+   $ pip install \
+     -e 'git+https://github.com/reanahub/reana-cluster.git@master#egg=reana-cluster'
+
+
+.. _configure:
+
+Configure REANA cluster
+-----------------------
+
+Main function of reana-cluster command line tool is to
+initialize a working REANA cluster, ready to run workflows
+you submit to it using reana-client.
+
+In order to achieve this reana-cluster needs to know how
+REANA cluster should be set up; e.g. what versions of REANA
+components should be deployed and how the configuration of each
+component should be set up.
+
+reana-client expects to get information via REANA cluster
```

Comment:
Isn't it `reana-cluster` expects to get that information via REANA cluster specification file?

Reply:
Yes, indeed.  This particular content comes from `master`, it's only restructured.  There will be some more updates coming, e.g. after we add `reana-cluster.yaml` in the package resources...

Can amend and repush anyway.

Reply:
Yes it is